### ISSUE TITLE
Set up optional asv runs

### ISSUE THREAD

**Issue by anissa111** [NCAR/geocat-comp#473]
Depends on #471 

**Description:**
Benchmarks should be able to be run on individual commits before they're merged into main when requested

**Comment by anissa111** [comment]
See https://github.com/anissa111/geocat-comp/pull/19 for working prototype

**Comment by erogluorhan** [comment]
> See [anissa111#19](https://github.com/anissa111/geocat-comp/pull/19) for working prototype

Perfect!
